zookeeper-server/src/test/java/org/apache/zookeeper/server/watch/WatcherCleanerTest.java

```diff
@@ -139,8 +141,10 @@ public void testMaxInProcessingDeadWatchers() {
         assertTrue(listener.wait(5000));
     }
 
-    @Test
-    public void testDeadWatcherMetrics() {
+    // There used to be a race condition surrounding this test which was reproducible by running the test multiple
+    // times. This test is kept as repeated to flag if the race condition reappears.
+    @RepeatedTest(5)
```

Comment:
I think it might be better to keep `@Test`. If this test is still flaky, it will fail finally. `@RepeatedTest` is a simple waste of resource.

Reply:
Sure, up to you